stan/math/prim/prob/beta_binomial_lpmf.hpp

```diff
@@ -159,7 +159,7 @@ return_type_t<T_size1, T_size2> beta_binomial_lpmf(const T_n& n, const T_N& N,
           += digamma_n_plus_alpha[i] + digamma_diff[i] - digamma_alpha[i];
     if (!is_constant_all<T_size2>::value)
       ops_partials.edge2_.partials_[i]
-          += digamma(value_of(N_vec[i] - n_vec[i] + beta_vec[i]))
+          += digamma(N_vec.val(i) - n_vec.val(i) + beta_vec.val(i))
```

Comment:
Note this is one place where I did fix something by hand. I this supposed to be taking these without val wrapped around them or was that a typo?

Reply:
This is a good fix. Doing the math before the value_of will create unnecessary varis if beta is autodiff.